### PR TITLE
Add secrets mnt to sidekiq

### PR DESCRIPTION
**Pull request by WillTaylor22**

### CODE REVIEW

File: charts/c100-application/templates/sidekiq.yaml [hunk 1]
```diff
@@ -46,9 +46,13 @@ spec:
             preStop:
               exec:
                 command: ["./sidekiq.sh", "stop"]
-          # non-secret env vars defined in `config_map.yaml`
+          {{ if .Values.global.enableKeyVaults }}
```
Comment (timja) [inline charts/c100-application/templates/sidekiq.yaml:49]: do you need this if?

File: charts/c100-application/values.dev.template.yaml [hunk 2]
```diff
@@ -30,4 +30,8 @@ redis:
     persistence:
       enabled: false
   auth:
-    enabled: false
\ No newline at end of file
+    enabled: false
+
+global:
+  enableKeyVaults: false
+  environment: dev
```
Comment (timja) [inline charts/c100-application/values.dev.template.yaml:37]: don't check this in, it's set by jenkins automatically with the correct env

File: charts/c100-application/values.stg.template.yaml [hunk 3]
```diff
@@ -2,3 +2,7 @@ base:
   # Don't modify below here
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+
+global:
+  enableKeyVaults: true
+  environment: stg
```
Comment (timja) [inline charts/c100-application/values.stg.template.yaml:8]: don't check this in

File: charts/c100-application/values.yaml [hunk 4]
```diff
@@ -45,3 +45,7 @@ base:
           alias: MAINTENANCE_ALLOWED_IPS
 redis:
   enabled: false
+useInterpodAntiAffinity: true
+keyVaults:
+  c100:
+    anything: here
```
Comment (timja) [inline charts/c100-application/values.yaml:51]: these should all be nested under base

File: charts/c100-application/values.yaml [hunk 5]
```diff
@@ -45,3 +45,7 @@ base:
           alias: MAINTENANCE_ALLOWED_IPS
 redis:
   enabled: false
+useInterpodAntiAffinity: true
```
Comment (timja) [inline charts/c100-application/values.yaml:48]: this is already enabled by default:
https://github.com/hmcts/chart-java/blob/8e0c2fcdac5ef416636e38e328ecb4de111f2d22/java/values.yaml#L32

File: charts/c100-application/templates/sidekiq.yaml (hunk 1 above)
Comment (WillTaylor22) [reply charts/c100-application/templates/sidekiq.yaml:49]: If not, the dev environment tries to build a keyvault and fails.

File: charts/c100-application/values.yaml (hunk 4 above)
Comment (WillTaylor22) [reply charts/c100-application/values.yaml:51]: "hmcts.interpodantiaffinity.v2"  requires useInterpodAntiAffinity at the root level.
"hmcts.secretCSIVolumes.v2" requires keyvaults: etc at the root level

File: charts/c100-application/values.yaml (hunk 5 above)
Comment (WillTaylor22) [reply charts/c100-application/values.yaml:48]: Without adding this here, "hmcts.interpodantiaffinity.v2" doesn't load.

File: charts/c100-application/values.dev.template.yaml (hunk 2 above)
Comment (WillTaylor22) [reply charts/c100-application/values.dev.template.yaml:37]: Needed for the IF statement above - can remove if the IF is not needed.

**Pull request merged**